frame/assets/src/impl_fungibles.rs

```diff
@@ -283,3 +283,12 @@ impl<T: Config<I>, I: 'static> fungibles::roles::Inspect<<T as SystemConfig>::Ac
 		Asset::<T, I>::get(asset).map(|x| x.freezer)
 	}
 }
+
+impl<T: Config<I>, I: 'static> fungibles::InspectEnumerable<T::AccountId> for Pallet<T, I> {
+	/// Returns an iterator of the assets in existence.
+	///
+	/// NOTE: iterating this list invokes a storage read per item.
+	fn asset_ids() -> Box<dyn Iterator<Item = Self::AssetId>> {
```

Comment:
What do you think about extracting the iterator type like it was done here: https://github.com/paritytech/substrate/pull/12389 ? That would remove the forced boxing.